### PR TITLE
rbd-mirror: replicate image metadata settings

### PR DESCRIPTION
**Pull request by trociny**
Fixes: http://tracker.ceph.com/issues/16212
Signed-off-by: Mykola Golub <mgolub@mirantis.com>

### CODE REVIEW

**Comment by trociny** [comment]
This needs #10980 to be merged first to avoid merge conflicts.

File: src/librbd/Operations.cc [hunk 1]
```diff
@@ -1297,6 +1299,115 @@ void Operations<I>::execute_update_features(uint64_t features, bool enabled,
...
+  size_t conf_prefix_len = start.size();
+
+  if (key.size() > conf_prefix_len && !key.compare(0, conf_prefix_len, start)) {
+    string subkey = key.substr(conf_prefix_len, key.size() - conf_prefix_len);
+    int r = cct->_conf->set_val(subkey.c_str(), value);
+    if (r < 0) {
+      return r;
+    }
+  }
+
```
Comment (dillaman) [inline src/librbd/Operations.cc]: Minor: check for m_image_ctx.read_only?

File: src/librbd/journal/Replay.cc [hunk 2]
```diff
@@ -689,6 +698,48 @@ void Replay<I>::handle_event(const journal::UpdateFeaturesEvent &event,
...
+  Context *on_op_complete = create_op_context_callback(event.op_tid, on_ready,
+                                                       on_safe, &op_event);
+  if (on_op_complete == nullptr) {
+    return;
+  }
+
+  op_event->on_op_finish_event = new C_RefreshIfRequired<I>(
+    m_image_ctx, new ExecuteOp<I, journal::MetadataRemoveEvent>(
+      m_image_ctx, event, on_op_complete));
+
```
Comment (dillaman) [inline src/librbd/journal/Replay.cc]: Minor: add "op_event->ignore_error_codes = {-ENOENT}" to ignore replay errors

**Comment by dillaman** [comment]
Minor: can you add new test cases to test_mock_Replay and test_Replay for the new events?

**Comment by trociny** [comment]
@dillaman Thanks for the comments. Updated.

**Comment by dillaman** [comment]
@trociny Still would like to see a journal/test_mock_Replay test case

**Comment by trociny** [comment]
@dillaman Rebased after #10980 merge.

**Pull request merged**